### PR TITLE
WIP: DF + Solo shuffles parser

### PR DESCRIPTION
**Pull request by prsauer**

### CODE REVIEW

File: packages/parser/src/actions/CombatAbsorbAction.ts [hunk 1]
```diff
@@ -21,10 +21,10 @@ export class CombatAbsorbAction extends CombatAction {
   constructor(logLine: ILogLine, wowVersion: WowVersion) {
     super(logLine);
     if (!CombatAbsorbAction.supports(logLine)) {
-      throw new Error('event not supported');
+      throw new Error('Event not supported as CombatAbsorbAction');
```
Comment (kunchenguid) [inline packages/parser/src/actions/CombatAbsorbAction.ts:24]: the error thrown from CombatAction.ts appends logLine.raw at the end. Should we consistently do so?

File: packages/parser/src/actions/CombatAdvancedAction.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export class CombatAdvancedAction extends CombatAction {
   constructor(logLine: ILogLine, wowVersion: WowVersion) {
     super(logLine);
     if (!CombatAdvancedAction.supports(logLine)) {
-      throw new Error('event not supported');
+      throw new Error('x event not supported');
```
Comment (kunchenguid) [inline packages/parser/src/actions/CombatAdvancedAction.ts:43]: what does x mean here? should we use a consistent format like in CombatAction.ts?

File: packages/parser/src/actions/CombatHpUpdateAction.ts [hunk 3]
```diff
@@ -3,15 +3,18 @@ import { CombatAdvancedAction } from './CombatAdvancedAction';
...
+    );
   }
 
   public readonly amount: number;
 
   constructor(logLine: ILogLine, wowVersion: WowVersion) {
     super(logLine, wowVersion);
     if (!CombatHpUpdateAction.supports(logLine)) {
-      throw new Error('event not supported');
+      throw new Error('Event not supported as CombatHpUpdateAction');
```
Comment (kunchenguid) [inline packages/parser/src/actions/CombatHpUpdateAction.ts:17]: same question as above - use a consistent format?

File: packages/parser/src/actions/PartyKill.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  public readonly destUnitName: string;
+  /**
+   * The killed unit
+   */
+  public readonly destUnitId: string;
+  public readonly destUnitFlags: number;
+
+  constructor(public readonly logLine: ILogLine) {
+    if (!PartyKill.supports(logLine)) {
+      throw new Error('event not supported');
```
Comment (kunchenguid) [inline packages/parser/src/actions/PartyKill.ts:31]: same question as above - use a consistent format?

File: packages/parser/src/actions/SpellAuraBrokenSpell.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    return super.supports(logLine) && logLine.event.endsWith('_BROKEN_SPELL');
+  }
+
+  public readonly breakingSpellId: string;
+  public readonly breakingSpellName: string;
+
+  constructor(logLine: ILogLine, wowVersion: WowVersion) {
+    super(logLine);
+    if (!SpellAuraBrokenSpell.supports(logLine)) {
+      throw new Error('Event not supported as SpellAuraBrokenSpell');
```
Comment (kunchenguid) [inline packages/parser/src/actions/SpellAuraBrokenSpell.ts:16]: same question as above - use a consistent format?

File: packages/parser/src/index.ts [hunk 6]
```diff
@@ -66,7 +66,7 @@ export class WoWCombatLogParser extends EventEmitter {
       }
 
       const wowBuild = wowVersionLineMatches[2];
-      const wowVersion: WowVersion = wowBuild.startsWith('2.') ? 'tbc' : 'retail';
+      const wowVersion: WowVersion = wowBuild.startsWith('2.') ? 'classic' : 'retail';
```
Comment (kunchenguid) [inline packages/parser/src/index.ts:69]: does wotlk start with "2."?

File: packages/parser/src/pipeline/retail/segmentToCombat.ts [hunk 7]
```diff
@@ -1,63 +1,231 @@
...
+// Global buffer to hold recent shuffle rounds
+// once a shuffle-ending is detected this is reset
+let recentShuffleRoundsBuffer: IShuffleRoundData[] = [];
+let recentScoreboardBuffer: IShuffleRoundData['scoreboard'] = {};
+
+function roundsBelongToSameMatch(roundA: ArenaMatchStartInfo, roundB: ArenaMatchStartInfo) {
+  // ARENA_MATCH_START is identical for every round of a shuffle except the timestamp
+  if (roundA.bracket !== roundB.bracket) return false;
+  if (roundA.isRanked !== roundB.isRanked) return false;
+  if (roundA.item1 !== roundB.item1) return false;
```
Comment (kunchenguid) [inline packages/parser/src/pipeline/retail/segmentToCombat.ts:21]: what is item1? 

does start info have combatants? i wonder if we should check for overlapping combatants as the best source of truth?

File: packages/shared/src/graphql-server/resolvers/profile.ts [hunk 8]
```diff
@@ -3,13 +3,13 @@ import { getUserProfileAsync } from '../utils/getUserProfileAsync';
 import { setUserReferrerAsync } from '../utils/setUserReferrerAsync';
 
 export async function me(parent: unknown, args: Record<string, unknown>, context: ApolloContext): Promise<User | null> {
-  return await getUserProfileAsync(context);
+  return getUserProfileAsync(context);
```
Comment (kunchenguid) [inline packages/shared/src/graphql-server/resolvers/profile.ts:6]: why remove await? I think we'll need to remove the async modifier on the line above if we do this.

File: packages/shared/src/graphql-server/resolvers/profile.ts [hunk 9]
```diff
@@ -3,13 +3,13 @@ import { getUserProfileAsync } from '../utils/getUserProfileAsync';
...
 }
 
 export async function setUserReferrer(
   _: unknown,
   args: { referrer: string | null },
   context: ApolloContext,
 ): Promise<User | null> {
-  return await setUserReferrerAsync(context, args.referrer);
+  return setUserReferrerAsync(context, args.referrer);
```
Comment (kunchenguid) [inline packages/shared/src/graphql-server/resolvers/profile.ts:14]: same question as above

File: packages/shared/src/graphql-server/resolvers/profile.ts (hunk 8 above)
Comment (prsauer) [reply packages/shared/src/graphql-server/resolvers/profile.ts:6]: some combo of eslint or prettier rules actually did this automatically, it gave some error that await wasn't valid in that context

File: packages/parser/src/pipeline/retail/segmentToCombat.ts (hunk 7 above)
Comment (prsauer) [reply packages/parser/src/pipeline/retail/segmentToCombat.ts:22]: item1 is unknown; startInfo doesnt have combatants (every field it has is used here except timestamp)

File: packages/parser/src/index.ts (hunk 6 above)
Comment (prsauer) [reply packages/parser/src/index.ts:69]: nope, its 3 -- good catch

File: packages/parser/src/actions/CombatAbsorbAction.ts (hunk 1 above)
Comment (prsauer) [reply packages/parser/src/actions/CombatAbsorbAction.ts:24]: yeah converting them all to a style that announces what event name failed and put the log line in there too

**Review by kunchenguid — APPROVED** [review]

**Pull request merged**